service/rtc/simulcast.go

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2022-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package rtc
+
+import (
+	"time"
+
+	"github.com/pion/interceptor/pkg/cc"
+
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+const (
+	SimulcastLevelHigh        = "h"
+	SimulcastLevelLow         = "l"
+	SimulcastLevelDefault     = SimulcastLevelLow
+	levelChangeInitialBackoff = 4 * time.Second
+	rateTolerance             = 0.9
+)
+
+var simulcastRates = []int{2_500_000, 500_000}
+var simulcastLevels = []string{SimulcastLevelHigh, SimulcastLevelLow}
+
+func getRateForSimulcastLevel(level string) int {
+	var rate int
+	for idx, lvl := range simulcastLevels {
+		if lvl == level {
+			rate = simulcastRates[idx]
+			break
+		}
+	}
+	return rate
+}
```

Comment:
nit: Do we forsee adding more levels in the near future? If not, would it be simpler to do a check and return? (up to you)

Reply:
Yeah. Originally we had three, which is why this logic is still more complex that how it needs to be. I can update it and if we ever want support more we implement it back.